### ISSUE TITLE
[BUG]: `print(String())` prints "(null)"

### ISSUE THREAD

**Issue by soraros** [modularml/mojo#1380]
### Bug description

The doc for `String.__init__(self)` says it constructs an empty string, while it does not.

### Steps to reproduce

```mojo
fn main():
  let s = String()
  print(s)  # prints (null)
```

### System information

```shell
Mojo 0.6.0 on Docker, Intel Mac
```

**Comment by JoeLoser** [comment]
`String.__init__(self)` is indeed explicitly constructing an empty buffer for its data it holds - that part is correct.  The problem was actually in the `print()` function itself.  This has been fixed internally, but just missed the cut-off for the `v0.6.1` release coming soon.  So, it will be included in the `v0.7.0` release in January sometime.  Thanks for the bug report!

**Comment by soraros** [comment]
@JoeLoser The doc string says "empty string" not empty buffer, if that difference is significant (null terminating or not?). Consider this example, I do think there is something wrong with `String()`:
```mojo
fn main():
    let s: String = ""
    print(s == "")  # prints True

    let t = String()
    print(t == "")  # prints False
```
